File: common/src/main/java/org/tdf/common/store/DatabaseStore.java
package org.tdf.common.store;

/**
 * Interface represents DB source which is normally the final Source in the chain
 */
public interface DatabaseStore extends BatchStore<byte[], byte[]> {
    /**
     * Initializes DB (open table, connection, etc)
     *
     * @param settings DB settings
     */
    void init(DBSettings settings);


    /**
     * @return true if DB connection is alive
     */
    boolean isAlive();

    /**
     * Closes the DB table/connection
     */
    void close();

    /**
     * Closes database, destroys its data and finally runs init()
     */
    void clear();
}

File: common/src/main/java/org/tdf/common/store/DatabaseStore.kt
package org.tdf.common.store

/**
 * Interface represents DB source which is normally the final Source in the chain
 */
interface DatabaseStore : BatchStore<ByteArray, ByteArray>, AutoCloseable {
    /**
     * Initializes DB (open table, connection, etc)
     *
     * @param settings DB settings
     */
    fun init(settings: DBSettings)

    /**
     * @return true if DB connection is alive
     */
    val alive: Boolean

    /**
     * Closes the DB table/connection
     */
    override fun close()

    /**
     * Closes database, destroys its data and finally runs init()
     */
    fun clear()
}